app/src/androidTest/java/de/test/antennapod/ui/DialogsTest.java

```diff
@@ -0,0 +1,98 @@
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.RootMatchers.isDialog;
+import static androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
+import static androidx.test.espresso.matcher.ViewMatchers.withClassName;
+import static androidx.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static de.test.antennapod.EspressoTestUtils.waitForView;
+import static de.test.antennapod.NthMatcher.first;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.endsWith;
+
+/**
+ * User interface tests for queue fragment.
+ */
+@RunWith(AndroidJUnit4.class)
+public class DialogsTest {
+
+    @Rule
+    public IntentsTestRule<MainActivity> activityRule = new IntentsTestRule<>(MainActivity.class, false, false);
+
+    @Before
+    public void setUp() throws InterruptedException {
+        EspressoTestUtils.clearPreferences();
+        EspressoTestUtils.clearDatabase();
+        EspressoTestUtils.setLastNavFragment(AddFeedFragment.TAG);
+        activityRule.launchActivity(new Intent());
+
+        String url = "https://omny.fm/shows/silence-is-not-an-option/why-not-being-racist-is-not-enough";
+
+        onView(withId(R.id.btn_add_via_url)).perform(scrollTo()).perform(click());
+        onView(withId(R.id.text)).perform(clearText(), typeText(url));
+        onView(withText(R.string.confirm_label)).inRoot(isDialog())
+                .check(matches(isDisplayed()))
+                .perform(closeSoftKeyboard())
+                .perform(scrollTo())
+                .perform(click());
+        Thread.sleep(5000);
+        onView(withId(R.id.butSubscribe)).perform(click());
+        Thread.sleep(5000);
+        onView(withId(R.id.recyclerView))
+                .perform(RecyclerViewActions.actionOnItemAtPosition(0, click()));
+        Thread.sleep(3000);
+        onView(first(EspressoTestUtils.actionBarOverflow())).perform(click());
+    }
+
+    @Test
+    public void testShareDialogDisplayed() {
+        onView(withText(R.string.share_label)).perform(scrollTo()).perform(click());
+        onView(allOf(isDisplayed(), withText(R.string.share_episode_label)));
+    }
+
+    @Test
+    public void testShareDialogShareButton() throws InterruptedException {
+        onView(withText(R.string.share_label)).perform(scrollTo()).perform(click());
+        onView(allOf(isDisplayed(), withText(R.string.share_label)));
+        Thread.sleep(1000);
+        onView(withText(R.string.share_episode_positive_label_button)).perform(scrollTo()).perform(click());
+        Thread.sleep(2000);
```

Comment:
Explicit sleep calls are usually not good for tests. It either makes them slow or it makes them fast but fails sometimes. Espresso should theoretically wait for the UI thread automatically.

Reply:
thanks I updated the tests :)